crates/cheatcodes/src/inspector.rs

```diff
@@ -1465,6 +1567,63 @@ impl Cheatcodes {
         }
     }
 
+    /// Generates arbitrary values for storage slots.
+    /// Invoked in inspector `step_end`, when the current opcode is not executed.
+    /// If current opcode to execute is `SLOAD` and storage slot is cold, then an arbitrary value
+    /// is generated and saved in target address storage (therefore when `SLOAD` opcode is executed,
+    /// the arbitrary value will be returned).
+    #[cold]
+    fn ensure_arbitrary_storage<DB: DatabaseExt>(
+        &mut self,
+        interpreter: &mut Interpreter,
+        ecx: &mut EvmContext<DB>,
+    ) {
+        if interpreter.current_opcode() == op::SLOAD {
```

Comment:
should move the `SLOAD` check outside into `step_end`, and make it call one `arbitrary_storage_end` to avoid callsite bloat

Reply:
changed in https://github.com/foundry-rs/foundry/pull/8807/commits/7e69723609d13ceb3ce744c0f7bbdd3e156f1296